authorization/invitation/service/invitation_service_blackbox_test.go

```diff
@@ -34,619 +38,693 @@ func (s *invitationServiceBlackBoxTest) SetupTest() {
+		// Create an admin role for the resource type
+		role, err := test.CreateTestRole(s.Ctx, s.DB, *resourceType, "admin")
+		require.NoError(s.T(), err)
 
-	invs, err := s.invitationRepo.ListForIdentity(s.Ctx, team.TeamID())
-	require.NoError(s.T(), err, "Error listing invitations")
+		// Assign the scope to our role
+		_, err = test.CreateTestRoleScope(s.Ctx, s.DB, *scope, *role)
+		require.NoError(s.T(), err)
 
-	require.Equal(s.T(), 2, len(invs))
+		// Create a resource
+		resource, err := test.CreateTestResource(s.Ctx, s.DB, *resourceType, "InvitationTestResourceFoo", nil)
+		require.NoError(s.T(), err)
 
-	found := false
+		// Assign the owner role to our user for the resource
+		test.CreateTestIdentityRoleForIdentity(s.Ctx, s.DB, identity, *resource, *role)
+		require.NoError(s.T(), err)
 
-	for _, inv := range invs {
-		if inv.IdentityID == invitee1.IdentityID() {
-			found = true
-			require.True(s.T(), inv.Member)
-			require.Equal(s.T(), invitee1.IdentityID(), inv.IdentityID)
-			require.Equal(s.T(), team.TeamID(), *inv.InviteTo)
+		// Create an invitation
+		invitations := []invitation.Invitation{
+			{
+				IdentityID: &otherIdentity.ID,
+				Member:     true,
+			},
 		}
-	}
 
-	require.True(s.T(), found, "First invitee not found in invitations")
+		// Issue the invitation, which should fail because the new resource can't have members
+		err = s.Application.InvitationService().Issue(s.Ctx, identity.ID, resource.ResourceID, invitations)
+		require.Error(s.T(), err)
+	})
+
+	s.T().Run("should fail to issue unprivileged invitation for resource", func(t *testing.T) {
+		// Create a test user - this will be the inviter
+		identity, err := test.CreateTestIdentityAndUserWithDefaultProviderType(s.DB, "invitationServiceBlackBoxTest-TestIssuingUser")
+		require.NoError(s.T(), err)
+
+		// Create another test user - we will invite this one to accept a role for the resource
+		otherIdentity, err := test.CreateTestIdentityAndUserWithDefaultProviderType(s.DB, "invitationServiceBlackBoxTest-TestInviteeUser")
+		require.NoError(s.T(), err)
+
+		// Create a new resource type
+		resourceType, err := test.CreateTestResourceType(s.Ctx, s.DB, "invitation.test/bar")
+		require.NoError(s.T(), err)
+
+		// Create an admin role for the resource type
+		_, err = test.CreateTestRole(s.Ctx, s.DB, *resourceType, "admin")
+		require.NoError(s.T(), err)
+
+		// Create a resource
+		resource, err := test.CreateTestResource(s.Ctx, s.DB, *resourceType, "InvitationTestResourceFoo", nil)
+		require.NoError(s.T(), err)
+
+		// Create an invitation
+		invitations := []invitation.Invitation{
+			{
+				IdentityID: &otherIdentity.ID,
+				Roles:      []string{"admin"},
+			},
+		}
+
+		// Issue the invitation, which should fail because the inviter has insufficient privileges to issue an invitation
+		err = s.Application.InvitationService().Issue(s.Ctx, identity.ID, resource.ResourceID, invitations)
+		require.Error(s.T(), err)
```

Comment:
same here: could you also verify that the NotificationService was not called at all, in this test?